public/js/lengthdistribution.js

```diff
@@ -15,16 +15,37 @@ export default class LengthDistribution extends React.Component {
   }
 
   componentDidMount() {
-    this.graph = new Graph(this.props.query, this.svgContainer(), this.props.algorithm)
+    this.graph = new Graph(this.props.query, this.svgContainer(), this.props.algorithm);
+    var svgContainer = this.svgContainer();
+    svgContainer
+    .on('mouseover', function () {
+      $(this).find('.hit-links').show();
+    })
+    .on('mouseleave', function() {
+      $(this).find('.hit-links').hide();
+    })
   }
 
   render() {
```

Comment:
I think you can take out the `render` method to a `Grapher` mixin as it will be the same for each graph (a daddy container with download buttons on top right and svg container to serve as canvas for drawing the graph.